dpctl/backend.pxd

```diff
@@ -67,27 +76,86 @@ cdef extern from "dppl_sycl_device_interface.h":
+    cdef bool DPPLProgram_HasKernel (DPPLSyclProgramRef PRef,
+                                     const char *KernelName)
+    cdef void DPPLProgram_Delete (DPPLSyclProgramRef PRef)
+
+
 cdef extern from "dppl_sycl_queue_interface.h":
-    cdef void DPPLQueue_Delete (DPPLSyclQueueRef QRef) except +
-    cdef DPPLSyclContextRef DPPLQueue_GetContext (const DPPLSyclQueueRef Q) \
-         except+
-    cdef DPPLSyclDeviceRef DPPLQueue_GetDevice (const DPPLSyclQueueRef Q) \
-         except +
+    cdef enum _arg_data_type 'DPPLKernelArgType':
+        _CHAR               'DPPL_CHAR',
+        _SIGNED_CHAR        'DPPL_SIGNED_CHAR',
+        _UNSIGNED_CHAR      'DPPL_UNSIGNED_CHAR',
+        _SHORT              'DPPL_SHORT',
+        _INT                'DPPL_INT',
+        _UNSIGNED_INT       'DPPL_INT',
+        _LONG               'DPPL_LONG',
+        _UNSIGNED_LONG      'DPPL_UNSIGNED_LONG',
+        _LONG_LONG          'DPPL_LONG_LONG',
+        _UNSIGNED_LONG_LONG 'DPPL_UNSIGNED_LONG_LONG',
+        _SIZE_T             'DPPL_SIZE_T',
+        _FLOAT              'DPPL_FLOAT',
+        _DOUBLE             'DPPL_DOUBLE',
+        _LONG_DOUBLE        'DPPL_DOUBLE',
+        _VOID_PTR           'DPPL_VOID_PTR'
+    ctypedef _arg_data_type DPPLKernelArgType
+    cdef void DPPLQueue_Delete (DPPLSyclQueueRef QRef)
+    cdef DPPLSyclContextRef DPPLQueue_GetContext (const DPPLSyclQueueRef Q)
+    cdef DPPLSyclDeviceRef DPPLQueue_GetDevice (const DPPLSyclQueueRef Q)
+    cdef DPPLSyclEventRef  DPPLQueue_SubmitRange (
+                                const DPPLSyclKernelRef Ref,
+                                const DPPLSyclQueueRef QRef,
+                                void **Args,
+                                const DPPLKernelArgType *ArgTypes,
+                                size_t NArgs,
+                                const size_t Range[3],
+                                size_t NDims,
+                                const DPPLSyclEventRef *DepEvents,
+                                size_t NDepEvents)
+    cdef DPPLSyclEventRef DPPLQueue_SubmitNDRange(
+                                const DPPLSyclKernelRef Ref,
+                                const DPPLSyclQueueRef QRef,
+                                void **Args,
+                                const DPPLKernelArgType *ArgTypes,
+                                size_t NArgs,
+                                const size_t gRange[3],
+                                const size_t lRange[3],
+                                size_t NDims,
+                                const DPPLSyclEventRef *DepEvents,
+                                size_t NDepEvents)
+    cdef void DPPLQueue_Wait (const DPPLSyclQueueRef QRef)
     cdef void DPPLQueue_Memcpy (const DPPLSyclQueueRef Q,
-                                void *Dest, const void *Src, size_t Count) \
-         except +
+                                void *Dest, const void *Src, size_t Count)
```

Comment:
@diptorupd as I remember you wanted to make Memcpy to return SyclEvent. In the next PR?

Reply:
Yes, I will open a follow up. Refer #84